Ers 434 merge or remove  identifier tables 2

robertkrantz opened the pull request.

psjobergprivat committed:
ERS-434 - Merging FluxReportDocumentEntity and _Identifier into FluxFaReportMessageEntity (they also have to be merged into FAReportDocument before being removed)

psjobergprivat committed:
Merge branch 'ERS-434-merge-or-remove-_identifier-tables' into ERS-434-merge-or-remove-_identifier-tables_2

psjobergprivat committed:
ERS-434 - Merging FluxReportDocumentEntity and _Identifier into FAReportDocument. Removing them. STILL COMPILATION ERRORS!

Commit:
Fixed mapping compilation problems.

There were two problems with the mappers. One was that there was no mapping for FaReportIdentifierEntity and RelatedReportDto. This was easily fixed by adding those methods to the BaseMapper class. The second was an ambiguous mapping error and this I'm not sure what the root problem was. A quick googling pointed to a problem with the target members being String and mapstruct not picking the right mapping methods. But since there were qualifiers on them it should have worked (?). The fix was instead to use the expression attribute and actually use the method we tried to point to with the qualifier.

Commit:
Fixed compilation errors.

Also had to fix my mapper fixes again, they didn't work apparently. There are still test failures which needs to be fixed.

Commit:
Fixed failing unit tests.

Had to change back some mapper changes due to NPE:s. Also fixed the related report mapper function in BaseMapper which I did a bit too fast without realising what it should be doing. The arquillian tests are still failing.

Commit:
Removed references to non-existing columns

Commit:
Merge branch 'swe-dev' of https://github.com/UnionVMS/UVMS-ActivityModule-APP into ERS-434-merge-or-remove-_identifier-tables_2

 Conflicts:
	pom.xml
	service/src/main/java/eu/europa/ec/fisheries/uvms/activity/fa/entities/FaReportDocumentEntity.java

Commit:
Added changesets for dropping foreign key constraints.

We tried to modify the data type of a column that was used for a foreign key constraint and also tried to drop a table that had a foreign key reference to it. Fixed a couple of queries.
Tests are still broken due to references to flux report document in several places.

Commit:
Updated search criteria builder to use the new purpose code column.

Since the flux report document entity was removed the query builder used the incorrect filter for purpose code. All unit tests now pass again. Still working on the arquillian tests.

Commit:
Fixed reference to fluxParty that still used the deleted flux report document table.

All tests now passes locally.

RobinMilding commented on service/src/main/java/eu/europa/ec/fisheries/uvms/activity/service/mapper/BaseMapper.java:
```diff
@@ -408,9 +409,21 @@ protected Instant dateTimeTypeToInstant(DateTimeType value) {
         return value.toGregorianCalendar().toInstant();
     }
+
+    @Named("OnlyInvokedByCustomMappers")
```
Is this name ever used? These names are referenced from @ Mapping annotations

RobinMilding commented on service/src/main/java/eu/europa/ec/fisheries/uvms/activity/service/bean/FluxMessageServiceBean.java:
```diff
@@ -218,18 +216,17 @@ private void enrichWithGuidFromAssets(VesselTransportMeansEntity vesselTransport
-            String justSavedPurposeCode = StringUtils.isNotEmpty(justSavedFluxReport.getPurposeCode()) ? justSavedFluxReport.getPurposeCode() : StringUtils.EMPTY;
+            String receivedRefId = justSavedReport.getFluxReportDocument_ReferenceId();
+            String receivedRefSchemeId = justSavedReport.getFluxReportDocument_ReferenceIdSchemeId();
+            String justSavedPurposeCode = StringUtils.isNotEmpty(justSavedReport.getFluxReportDocument_PurposeCode()) ?
```
Is this just to avoid a possible null value?

RobinMilding commented on service/src/main/java/eu/europa/ec/fisheries/uvms/activity/service/mapper/BaseMapper.java:
```diff
@@ -430,4 +432,33 @@ protected String singleIDTypeSchemeID(List<IDType> idTypes) {
+
+    private Set<FaReportIdentifierEntity> mapRelatedReportIDs(FAReportDocument target) {
+        Set<FaReportIdentifierEntity> faReportIdentifiers = new HashSet<>();
+        if (CollectionUtils.isNotEmpty(faReportIdentifiers)) {
```
How can this not be empty when you just created it?

RobinMilding commented on service/src/test/java/eu/europa/ec/fisheries/uvms/activity/service/mapper/ActivityEntityToModelMapperTest.java:
```diff
@@ -65,6 +65,10 @@ public void testMapToFLUXFAReportMessage(String resource) throws Exception {
         String clearControlSource = clearEmptyTags(controlSource);
         String clearTestSource = clearEmptyTags(testSource);
 
+        System.out.println(clearControlSource);
```
Forgotton dev output?

RobinMilding commented on service/src/main/java/eu/europa/ec/fisheries/uvms/activity/service/search/FilterMap.java:
```diff
@@ -232,7 +230,7 @@ public void populateFilterMappings() {
         filterMappings.put(SearchFilter.VESSEL_IDENTIFIRE, new FilterDetails("vt.vesselIdentifiers vi", "vi.vesselIdentifierId IN (:" + VESSEL_IDENTIFIRE + ")"));
         filterMappings.put(SearchFilter.VESSEL_GUIDS, new FilterDetails("fa.vesselTransportMeans vtMeans", "vtMeans.guid IN (:" + VTM_GUIDS + ")"));
-        filterMappings.put(SearchFilter.PURPOSE, new FilterDetails(FLUX_REPORT_DOC_TABLE_ALIAS, "flux.purposeCode IN (:" + PURPOSE_CODE + ")"));
+        filterMappings.put(SearchFilter.PURPOSE, new FilterDetails(" ", "fa.fluxReportDocument_PurposeCode IN (:" + PURPOSE_CODE + ")"));
```
The String-with-only-a-space looks funny but I'm not sure what it does so it  might be correct

RobinMilding commented on service/src/main/java/eu/europa/ec/fisheries/uvms/activity/service/search/builder/SearchQueryBuilder.java:
```diff
@@ -195,42 +193,21 @@ private StringBuilder getJoinPartForSortingOptions(StringBuilder sql, FishingAct
-        }
-    }
-
     private int getFiledCase(StringBuilder sql, SearchFilter field) {
```
Return boolean instead?

RobinMilding commented on service/src/test/java/eu/europa/ec/fisheries/uvms/activity/fa/entities/SearchQueryBuilderTest.java:
```diff
@@ -98,8 +98,47 @@ public void createSQL_DateSorting() throws ServiceException {
         // When
         StringBuilder sql = search.createSQL(query);
 
+        String expected =
```
Nice

Commit:
Removed commented and unused methods.

Also removed Named annotation from method only used by other methods in the same class. There was no reference in any mapper.

Commit:
Removed unnecessary method

Commit:
Merge branch 'swe-dev' of https://github.com/UnionVMS/UVMS-ActivityModule-APP into ERS-434-merge-or-remove-_identifier-tables_2

 Conflicts:
	service/src/test/java/eu/europa/ec/fisheries/uvms/activity/fa/dao/FluxReportIdentifierDaoTest.java
	service/src/test/java/eu/europa/ec/fisheries/uvms/activity/service/mapper/FluxReportDocumentMapperTest.java
